man/just.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.49.3.
.TH JUST "1" "March 2024" "just 1.25.2" "Just Manual"
.SH NAME
just \- save and run commands
.SH DESCRIPTION
just 1.25.2
\- Please see https://github.com/casey/just for more information.
.SS "USAGE:"
.IP
just [FLAGS] [OPTIONS] [\-\-] [ARGUMENTS]...
.SS "FLAGS:"
.TP
\fB\-\-changelog\fR
Print changelog
.TP
\fB\-\-check\fR
Run `\-\-fmt` in 'check' mode. Exits with 0 if justfile is formatted
correctly. Exits with 1 and prints a diff if formatting is required.
.TP
\fB\-\-choose\fR
Select one or more recipes to run using a binary chooser. If `\-\-chooser` is
not passed the chooser defaults to the value of $JUST_CHOOSER, falling back
to `fzf`
.TP
\fB\-\-clear\-shell\-args\fR
Clear shell arguments
.TP
\fB\-n\fR, \fB\-\-dry\-run\fR
Print what just would do without doing it
.TP
\fB\-\-dump\fR
Print justfile
.TP
\fB\-e\fR, \fB\-\-edit\fR
Edit justfile with editor given by $VISUAL or $EDITOR, falling back to
`vim`
.TP
\fB\-\-evaluate\fR
Evaluate and print all variables. If a variable name is given as an
argument, only print that variable's value.
.TP
\fB\-\-fmt\fR
Format and overwrite justfile
.TP
\fB\-\-highlight\fR
Highlight echoed recipe lines in bold
.TP
\fB\-\-init\fR
Initialize new justfile in project root
.TP
\fB\-l\fR, \fB\-\-list\fR
List available recipes and their arguments
.TP
\fB\-\-no\-deps\fR
Don't run recipe dependencies
.TP
\fB\-\-no\-dotenv\fR
Don't load `.env` file
.TP
\fB\-\-no\-highlight\fR
Don't highlight echoed recipe lines in bold
.TP
\fB\-q\fR, \fB\-\-quiet\fR
Suppress all output
.TP
\fB\-\-shell\-command\fR
Invoke <COMMAND> with the shell used to run recipe lines and backticks
.TP
\fB\-\-summary\fR
List names of available recipes
.TP
\fB\-u\fR, \fB\-\-unsorted\fR
Return list and summary entries in source order
.TP
\fB\-\-unstable\fR
Enable unstable features
.TP
\fB\-\-variables\fR
List names of variables
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Use verbose output
.TP
\fB\-\-yes\fR
Automatically confirm all recipes.
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help information
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version information
.SS "OPTIONS:"
.TP
\fB\-\-chooser\fR <CHOOSER>
Override binary invoked by `\-\-choose`
.HP
\fB\-\-color\fR <COLOR>
.TP
Print colorful output [default: auto]
[possible values: auto, always, never]
.HP
\fB\-c\fR, \fB\-\-command\fR <COMMAND>
.IP
Run an arbitrary command with the working directory, `.env`, overrides, and exports set
.HP
\fB\-\-command\-color\fR <COMMAND\-COLOR>
.IP
Echo recipe lines in <COMMAND\-COLOR> [possible values: black, blue, cyan, green, purple, red,
yellow]
.HP
\fB\-\-completions\fR <SHELL>
.IP
Print shell completion script for <SHELL> [possible values: zsh, bash, fish, powershell,
elvish]
.HP
\fB\-\-dotenv\-filename\fR <DOTENV\-FILENAME>
.IP
Search for environment file named <DOTENV\-FILENAME> instead of `.env`
.HP
\fB\-\-dotenv\-path\fR <DOTENV\-PATH>
.IP
Load environment file at <DOTENV\-PATH> instead of searching for one
.HP
\fB\-\-dump\-format\fR <FORMAT>
.TP
Dump justfile as <FORMAT> [default: just]
[possible values: just, json]
.TP
\fB\-f\fR, \fB\-\-justfile\fR <JUSTFILE>
Use <JUSTFILE> as justfile
.TP
\fB\-\-list\-heading\fR <TEXT>
Print <TEXT> before list
.TP
\fB\-\-list\-prefix\fR <TEXT>
Print <TEXT> before each list item
.TP
\fB\-\-set\fR <VARIABLE> <VALUE>
Override <VARIABLE> with <VALUE>
.TP
\fB\-\-shell\fR <SHELL>
Invoke <SHELL> to run recipes
.TP
\fB\-\-shell\-arg\fR <SHELL\-ARG>...
Invoke shell with <SHELL\-ARG> as an argument
.TP
\fB\-s\fR, \fB\-\-show\fR <RECIPE>
Show information about <RECIPE>
.HP
\fB\-d\fR, \fB\-\-working\-directory\fR <WORKING\-DIRECTORY>
.IP
Use <WORKING\-DIRECTORY> as working directory. \fB\-\-justfile\fR must also be set
.SS "ARGS:"
.TP
<ARGUMENTS>...
Overrides and recipe(s) to run, defaulting to the first recipe in the justfile


CHANGELOG.md
Changelog
=========

[1.25.2](https://github.com/casey/just/releases/tag/1.25.2) - 2024-03-10
------------------------------------------------------------------------

- Unpin ctrlc ([#1951](https://github.com/casey/just/pull/1951))

[1.25.1](https://github.com/casey/just/releases/tag/1.25.1) - 2024-03-09
------------------------------------------------------------------------

### Misc
- Pin ctrlc to version 3.1.1 ([#1945](https://github.com/casey/just/pull/1945))
- Fix AArch64 release build error ([#1942](https://github.com/casey/just/pull/1942))

[1.25.0](https://github.com/casey/just/releases/tag/1.25.0) - 2024-03-07
------------------------------------------------------------------------

### Added
- Add `blake3` and `blake3_file` functions ([#1860](https://github.com/casey/just/pull/1860) by [tgross35](https://github.com/tgross35))

### Misc
- Fix readme typo ([#1936](https://github.com/casey/just/pull/1936) by [Justintime50](https://github.com/Justintime50))
- Use unwrap_or_default ([#1928](https://github.com/casey/just/pull/1928))
- Set codegen-units to 1 reduce release binary size ([#1920](https://github.com/casey/just/pull/1920) by [amarao](https://github.com/amarao))
- Document openSUSE package ([#1918](https://github.com/casey/just/pull/1918) by [sfalken](https://github.com/sfalken))
- Fix install.sh shellcheck warnings ([#1912](https://github.com/casey/just/pull/1912) by [tgross35](https://github.com/tgross35))

[1.24.0](https://github.com/casey/just/releases/tag/1.24.0) - 2024-02-11
------------------------------------------------------------------------

### Added
- Support recipe paths containing `::` in Bash completion script ([#1863](https://github.com/casey/just/pull/1863) by [crdx](https://github.com/crdx))
- Add function to canonicalize paths ([#1859](https://github.com/casey/just/pull/1859))

### Misc
- Document installing just on Github Actions in readme ([#1867](https://github.com/casey/just/pull/1867) by [cclauss](https://github.com/cclauss))
- Use unlikely-to-be-set variable name in env tests ([#1882](https://github.com/casey/just/pull/1882))
- Skip write_error test if running as root ([#1881](https://github.com/casey/just/pull/1881))
- Convert run_shebang into integration test ([#1880](https://github.com/casey/just/pull/1880))
- Install mdbook with cargo in CI workflow ([#1877](https://github.com/casey/just/pull/1877))
- Remove deprecated actions-rs/toolchain ([#1874](https://github.com/casey/just/pull/1874) by [cclauss](https://github.com/cclauss))
- Fix Gentoo package link ([#1875](https://github.com/casey/just/pull/1875) by [vozbu](https://github.com/vozbu))
- Fix typos found by codespell ([#1872](https://github.com/casey/just/pull/1872) by [cclauss](https://github.com/cclauss))
- Replace deprecated set-output command in Github Actions workflows ([#1869](https://github.com/casey/just/pull/1869) by [cclauss](https://github.com/cclauss))
- Update `actions/checkout` and `softprops/action-gh-release` ([#1871](https://github.com/casey/just/pull/1871) by [app/dependabot](https://github.com/app/dependabot))
- Keep GitHub Actions up to date with Dependabot ([#1868](https://github.com/casey/just/pull/1868) by [cclauss](https://github.com/cclauss))
- Add contrib directory ([#1870](https://github.com/casey/just/pull/1870))
- Fix install script ([#1844](https://github.com/casey/just/pull/1844))

[1.23.0](https://github.com/casey/just/releases/tag/1.23.0) - 2024-01-12
------------------------------------------------------------------------

### Added
- Allow setting custom confirm prompt ([#1834](https://github.com/casey/just/pull/1834) by [CramBL](https://github.com/CramBL))
- Add `set quiet` and `[no-quiet]` ([#1704](https://github.com/casey/just/pull/1704) by [dharrigan](https://github.com/dharrigan))
- Add `just_pid` function ([#1833](https://github.com/casey/just/pull/1833) by [Swordelf2](https://github.com/Swordelf2))
- Add functions to return XDG base directories ([#1822](https://github.com/casey/just/pull/1822) by [tgross35](https://github.com/tgross35))
- Add `--no-deps` to skip running recipe dependencies ([#1819](https://github.com/casey/just/pull/1819) by [ngharrington](https://github.com/ngharrington))

### Fixed
- Run imports in working directory of importer ([#1817](https://github.com/casey/just/pull/1817))

### Misc
- Include completion scripts in releases ([#1837](https://github.com/casey/just/pull/1837))
- Tweak readme table formatting ([#1836](https://github.com/casey/just/pull/1836))
- Don't abbreviate just in README ([#1831](https://github.com/casey/just/pull/1831) by [thled](https://github.com/thled))
- Ignore [private] recipes in just --list ([#1816](https://github.com/casey/just/pull/1816) by [crdx](https://github.com/crdx))
- Add a dash to tempdir prefix ([#1828](https://github.com/casey/just/pull/1828))

[1.22.1](https://github.com/casey/just/releases/tag/1.22.1) - 2024-01-08
------------------------------------------------------------------------

### Fixed
- Don't conflate recipes with the same name in different modules ([#1825](https://github.com/casey/just/pull/1825))

### Misc
- Clarify that UUID is version 4 ([#1821](https://github.com/casey/just/pull/1821) by [tgross35](https://github.com/tgross35))
- Make sigil stripping from recipe lines less incomprehensible ([#1812](https://github.com/casey/just/pull/1812))
- Refactor invalid path argument check ([#1811](https://github.com/casey/just/pull/1811))

[1.22.0](https://github.com/casey/just/releases/tag/1.22.0) - 2023-12-31
------------------------------------------------------------------------

### Added
- Recipes can be invoked with path syntax ([#1809](https://github.com/casey/just/pull/1809))
- Add `--format` and `--initialize` as aliases for `--fmt` and `--init` ([#1802](https://github.com/casey/just/pull/1802))

### Misc
- Move table of contents pointer to right ([#1806](https://github.com/casey/just/pull/1806))


completions/just.bash
_just() {
    local i cur prev opts cmds
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${i}" in
            "$1")
                cmd="just"
                ;;
            
            *)
                ;;
        esac
    done

    case "${cmd}" in
        just)
            opts=" -n -p -q -u -v -e -l -h -V -f -d -c -s  --check --yes --dry-run --highlight --no-dotenv --no-highlight --parallel --quiet --shell-command --clear-shell-args --unsorted --unstable --verbose --changelog --choose --dump --edit --evaluate --fmt --init --list --summary --variables --help --version --chooser --color --command-color --dump-format --list-heading --list-prefix --justfile --set --shell --shell-arg --working-directory --command --completions --show --dotenv-filename --dotenv-path  <ARGUMENTS>... "
                if [[ ${cur} == -* ]] ; then
                    COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                    return 0
                elif [[ ${COMP_CWORD} -eq 1 ]]; then
                    local recipes=$(just --summary 2> /dev/null)

                    if echo "${cur}" | \grep -qF '/'; then
                        local path_prefix=$(echo "${cur}" | sed 's/[/][^/]*$/\//')
                        local recipes=$(just --summary 2> /dev/null -- "${path_prefix}")
                        local recipes=$(printf "${path_prefix}%s\t" $recipes)
                    fi

                    if [[ $? -eq 0 ]]; then
                        COMPREPLY=( $(compgen -W "${recipes}" -- "${cur}") )
                        return 0
                    fi
                fi
            case "${prev}" in
                
                --chooser)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --color)
                    COMPREPLY=($(compgen -W "auto always never" -- "${cur}"))
                    return 0
                    ;;
                --command-color)
                    COMPREPLY=($(compgen -W "black blue cyan green purple red yellow" -- "${cur}"))
                    return 0
                    ;;
                --dump-format)
                    COMPREPLY=($(compgen -W "just json" -- "${cur}"))
                    return 0
                    ;;
                --list-heading)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --list-prefix)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --justfile)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                    -f)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --set)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --shell)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --shell-arg)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --working-directory)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                    -d)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --command)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                    -c)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --completions)
                    COMPREPLY=($(compgen -W "zsh bash fish powershell elvish" -- "${cur}"))
                    return 0
                    ;;
                --show)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                    -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --dotenv-filename)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --dotenv-path)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        
    esac
}

complete -F _just -o bashdefault -o default just
